lusidtools/commands/commands.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import sys
+import lusidtools.lpt as lpt
+import os
+import ast
+
+
+def find_tools():
+    root = lpt.__path__[0]
+    for filename in os.listdir(root):
+        if filename.endswith('.py'):
+            with open(os.path.join(root, filename), "rt") as file:
+                toolname = filename[:-3]
+                toolmodule = toolname
+                tooltip = ""
+                tree = ast.parse(file.read(), filename=filename)
+                for f in tree.body:
+                    if isinstance(f, ast.Assign):
+                        if (len(f.targets) == 1 and
+                                isinstance(f.targets[0], ast.Name) and
+                                isinstance(f.value, ast.Str)):
+                            if f.targets[0].id == "TOOLNAME":
+                                toolname = f.value.s
+                            elif f.targets[0].id == "TOOLTIP":
+                                tooltip = f.value.s
+                    if isinstance(f, ast.FunctionDef):
+                        if f.name == 'main':
+                            yield toolname, (toolmodule, tooltip)
+
+
+def help_method(args, tools):
+    def add_tip(tool, tip):
+        return tool if tip == '' else f'{tool: <30} - {tip}'
+
+    epilog = (
+            "Available tools. - use <tool> -h for additional help\n\n" +
+            "\n".join(sorted([add_tip(k, v[1]) for k, v in tools.items()]))
+    )
+
+    parser = argparse.ArgumentParser(
+        description="Lusid Python Tools Suite",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog=epilog
+    )
+    parser.add_argument('tool', metavar='<tool>', help='Tool to run. Options are listed below.')
+    parser.add_argument('arg1', metavar='<arg-1>', help=' - tool')
+    parser.add_argument('arg2', metavar=' ... ', help=' - specific')
+    parser.add_argument('arg3', metavar='<arg-n>', help=' - arguments')
+    parser.parse_args(['--help'])
+    exit()
+
+
+def main():
+    tools = dict(find_tools())
```

Comment:
What is the shape of this dictionary given it comes from a generator?

Reply:
```
{
    "qry_holdings": ("hld", "gets holdings of a portfolio")
}
```